charts/internal/seed-controlplane/charts/cloud-controller-manager/templates/cloud-controller-manager.yaml

```diff
@@ -37,7 +37,7 @@ spec:
       - name: gcp-cloud-controller-manager
         image: {{ index .Values.images "cloud-controller-manager" }}
         imagePullPolicy: IfNotPresent
-        {{- if semverCompare ">= 1.31" .Values.kubernetesVersion }}
+        {{- if semverCompare "~ 1.31.x-0" .Values.kubernetesVersion }}
```

Comment:
This is copied from https://github.com/gardener/gardener/blob/9f78a779c44ae2831ea2459b12a7c1df499ca613/pkg/utils/version/version.go#L52.

You could use 
```
% helm template -s templates/cloud-controller-manager.yaml ./charts/internal/seed-controlplane/charts/cloud-controller-manager --set global.genericTokenKubeconfigSecretNam=foo --set kubernetesVersion=1.31.0
```

to verify that it is working as expected.